packages-exp/performance-exp/src/services/transport_service.test.ts

```diff
@@ -82,32 +84,102 @@ describe('Firebase Performance > transport_service', () => {
   });
 
   it('successful send a meesage to transport', () => {
-    const transportDelayInterval = 30000;
     const setting = SettingsService.getInstance();
     const flTransportFullUrl =
       setting.flTransportEndpointUrl + '?key=' + setting.transportKey;
     fetchStub.withArgs(flTransportFullUrl, match.any).resolves(
       // DELETE_REQUEST means event dispatch is successful.
-      new Response(
-        '{\
-        "nextRequestWaitMillis": "' +
-          transportDelayInterval +
-          '",\
-        "logResponseDetails": [\
-          {\
-            "responseAction": "DELETE_REQUEST"\
-          }\
-        ]\
-      }',
-        {
-          status: 200,
-          headers: { 'Content-type': 'application/json' }
-        }
-      )
+      generateSuccessResponse()
     );
 
     testTransportHandler('event1');
     clock.tick(INITIAL_SEND_TIME_DELAY_MS);
     expect(fetchStub).to.have.been.calledOnce;
   });
+
+  it('sends up to the maximum event limit in one request', async () => {
+    // Arrange
+    const setting = SettingsService.getInstance();
+    const flTransportFullUrl =
+      setting.flTransportEndpointUrl + '?key=' + setting.transportKey;
+    // Generates the first logRequest which contains first 1000 events.
+    const logRequest = generateLogRequest('5501');
+    for (let i = 0; i < MAX_EVENT_COUNT_PER_REQUEST; i++) {
+      logRequest['log_event'].push({
+        'source_extension_json_proto3': 'event' + i,
+        'event_time_ms': '1'
+      });
+    }
+
+    // Returns successful response from fl for logRequests.
+    const response = generateSuccessResponse();
+    fetchStub
+      .withArgs(flTransportFullUrl, {
+        method: 'POST',
+        body: JSON.stringify(logRequest)
+      })
+      .resolves(response);
+    stub(response, 'json').returns(
+      Promise.resolve(JSON.parse(generateSuccessResponseBody()))
+    );
+
+    // Act
+    // Generate 1020 events, which should be dispatched in two batches (1000 events and 20 events).
+    for (let i = 0; i < 1020; i++) {
+      testTransportHandler('event' + i);
+    }
+
+    // Assert
+    clock.tick(INITIAL_SEND_TIME_DELAY_MS);
+    // First logRequest has been called.
+    expect(fetchStub).to.have.been.calledWith(flTransportFullUrl, {
+      method: 'POST',
+      body: JSON.stringify(logRequest)
+    });
+    // Wait for async action to call for next dispatch cycle.
+    await Promise.resolve()
+      .then(() => {
+        clock.tick(1);
+      })
+      .then(() => {
+        clock.tick(1);
+      });
+    clock.tick(DEFAULT_SEND_INTERVAL_MS * 3);
+    // Remaining events stay in queue.
+    expect(readQueue().length).to.be.equal(20);
```

Comment:
Shouldn't the 20 logs be sent in the next batch?

Reply:
That is correct, switched test logic to validate two batches with 1000 and 20 events each.